File: demo/ffi/ydraw/lua/yimage/gallery.lua
-- ydraw client interface target sketch — yimage complex drawables (Lua).
-- NOT RUNNABLE YET — see python/yimage/gallery.py. Asset paths are
-- repo-relative. Assets: demo/assets/yimage/.
local ydraw = require("yetty.ydraw")

local ASSETS = "demo/assets/yimage/"

local function show(...)
    local dlist = ydraw.DrawableList()
    for _, drawable in ipairs({...}) do
        dlist:add(drawable)
    end
    dlist:dcs_emit()
    dlist:destroy()
end

print("rose")
show(ydraw.Image{ASSETS .. "rose.png"})

print("hero, scaled to 480px wide")
show(ydraw.Image{ASSETS .. "hero.png", width = 480, height = 270})

-- One record among others: caption in the same envelope.
print("wordmark with caption")
show(ydraw.Image{ASSETS .. "wordmark.png", x = 0, y = 0, width = 320, height = 96},
     ydraw.Text{"terminal unchained", x = 0, y = 112, font_size = 18,
                color = "#9FA7A8"})

print("thumbnail strip")
local strip = ydraw.DrawableList()
local names = {"gradient.png", "rose.png", "hero.png", "wordmark.png"}
for index, name in ipairs(names) do
    strip:add(ydraw.Image{ASSETS .. name, x = (index - 1) * 170, y = 0,
                          width = 160, height = 100})
end
strip:dcs_emit()
strip:destroy()

File: demo/ffi/ydraw/typescript/yimage/gallery.ts
// ydraw client interface target sketch — yimage complex drawables (TS).
// NOT RUNNABLE YET — see python/yimage/gallery.py. Asset paths are
// repo-relative. Assets: demo/assets/yimage/.
import { Drawable, DrawableList, Image, Text } from "@yetty/ydraw";

const ASSETS = "demo/assets/yimage/";

function show(...drawables: Drawable[]): void {
  const dlist = new DrawableList();
  for (const drawable of drawables) {
    dlist.add(drawable);
  }
  dlist.dcsEmit();
  dlist.destroy();
}

console.log("rose");
show(new Image(ASSETS + "rose.png"));

console.log("hero, scaled to 480px wide");
show(new Image(ASSETS + "hero.png", { width: 480, height: 270 }));

// One record among others: caption in the same envelope.
console.log("wordmark with caption");
show(new Image(ASSETS + "wordmark.png", { x: 0, y: 0, width: 320, height: 96 }),
     new Text("terminal unchained", { x: 0, y: 112, fontSize: 18,
                                      color: "#9FA7A8" }));

console.log("thumbnail strip");
const strip = new DrawableList();
["gradient.png", "rose.png", "hero.png", "wordmark.png"].forEach((name, index) => {
  strip.add(new Image(ASSETS + name, { x: index * 170, y: 0,
                                       width: 160, height: 100 }));
});
strip.dcsEmit();
strip.destroy();
